internal/collector/settings_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+package collector
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/nginx/agent/v3/internal/config"
+	"github.com/nginx/agent/v3/test/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOTelCollectorSettings(t *testing.T) {
+	cfg := types.GetAgentConfig()
+
+	settings := OTelCollectorSettings(cfg)
+
+	assert.NotNil(t, settings.Factories, "Factories should not be nil")
+	assert.Equal(t, "otel-nginx-agent", settings.BuildInfo.Command, "BuildInfo command should match")
+	assert.False(t, settings.DisableGracefulShutdown, "DisableGracefulShutdown should be false")
+	assert.NotNil(t, settings.ConfigProviderSettings, "ConfigProviderSettings should not be nil")
+	assert.Nil(t, settings.LoggingOptions, "LoggingOptions should be nil")
+	assert.False(t, settings.SkipSettingGRPCLogger, "SkipSettingGRPCLogger should be false")
+}
+
+func TestConfigProviderSettings(t *testing.T) {
+	cfg := types.GetAgentConfig()
+	settings := ConfigProviderSettings(cfg)
+
+	assert.NotNil(t, settings.ResolverSettings, "ResolverSettings should not be nil")
+	assert.Len(t, settings.ResolverSettings.ProviderFactories, 5, "There should be 5 provider factories")
+	assert.Len(t, settings.ResolverSettings.ConverterFactories, 1, "There should be 1 converter factory")
+	assert.NotEmpty(t, settings.ResolverSettings.URIs, "URIs should not be empty")
+	assert.Equal(t, "/etc/nginx-agent/nginx-agent-otelcol.yaml", settings.ResolverSettings.URIs[0],
+		"Default URI should match")
+}
+
+func TestTemplateWrite(t *testing.T) {
+	cfg := types.GetAgentConfig()
+	actualConfPath := filepath.Join(t.TempDir(), "nginx-agent-otelcol-test.yaml")
+	cfg.Collector.ConfigPath = actualConfPath
+	// cfg.Collector.ConfigPath = "testdata/test-otelcol.yaml"
+
+	cfg.Collector.Exporters = append(cfg.Collector.Exporters, config.Exporter{
```

Comment:
if you do https://github.com/nginx/agent/pull/698/files#r1636799918 or the next comment you can call GetConfiig().Exporters then and reuse the centralised config here and in the below lines

Reply:
I'd like to keep this as is, so that the version in `test/types/config.go` has a default configuration, which will be more sensible for integration tests. Here we are going for the "all out" configuration with all possible supported exporters, producers and receivers.